incubator/hnc/internal/pkg/selectors/selectors.go

```diff
@@ -61,23 +61,35 @@ func GetTreeSelector(inst *unstructured.Unstructured) (labels.Selector, error) {
 
 	segs := strings.Split(treeSelectorStr, ",")
 	selectorStr := ""
+	nonNegatedNses := []string{}
```

Comment:
Rather than building a list, you could just early-return as soon as the _second_ non-negated name is found.

Reply:
Do we want to include the non-negated nses in the error msg? Cause if we do then a list seems to be a better option